packages/unified-selection/src/unified-selection/HiliteSetProvider.ts

```diff
@@ -0,0 +1,324 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+import { ECClass, IMetadataProvider, parseFullClassName } from "./queries/ECMetadata";
+import { ECSqlBinding, IECSqlQueryExecutor } from "./queries/ECSqlCore";
+import { SelectableInstanceKey, Selectables } from "./Selectable";
+
+/**
+ * A set of model, subcategory and element ids that can be used for specifying hilite.
+ * @beta
+ */
+export interface HiliteSet {
+  models: string[];
+  subCategories: string[];
+  elements: string[];
+}
+
+/**
+ * Props for creating a `HiliteSetProvider` instance.
+ * @beta
+ */
+export interface HiliteSetProviderProps {
+  queryExecutor: IECSqlQueryExecutor;
+  metadataProvider: IMetadataProvider;
+}
+
+/**
+ * ECSQL based provider which determines what `HiliteSet`
+ * should be hilited based on the supplied `Selectables`.
+ * @beta
+ */
+export interface HiliteSetProvider {
+  /** Get the current hilite set iterator for the specified imodel */
+  getHiliteSet(props: { selectables: Selectables }): AsyncIterableIterator<HiliteSet>;
+}
+
+/**
+ * Creates a hilite set provider.
+ * @beta
+ */
+export function createHiliteSetProvider(props: HiliteSetProviderProps): HiliteSetProvider {
+  return new HiliteSetProviderImpl(props);
+}
+
+class HiliteSetProviderImpl implements HiliteSetProvider {
```

Comment:
Looks like this is used only internally. In that case I think we don't need `HiliteSetProvider` interface and `createHiliteSetProvider` function. We can use `HiliteSetProviderImpl` class directly in `CachingHiliteSetProvider`.

Reply:
`HiliteSetProvider` was supposed to be exported, fixed.